src/screens/app/screens/AddTask/screens/ChooseTaskDifficulty.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext, useState, useCallback } from 'react'
+import { StyleSheet, ScrollView } from 'react-native'
+import { useNavigation, useRoute, useFocusEffect } from '@react-navigation/core'
+
+import { useT } from 'res/i18n'
+
+import { FormLayout } from 'library/layouts/FormLayout'
+import { Label } from 'library/components/Label'
+import { UserTaskContext } from 'screens/app/contexts/UserTask.context'
+import { TaskDifficulty } from './components/TaskDifficulty'
+import { TaskAddedModalContext } from '../../Dashboard/components/TaskAddedModal'
+
+export const ChooseTaskDifficultyScreen = () => {
+  const { t } = useT()
+  const route = useRoute()
+  const navigation = useNavigation()
+  const [selectedIndex, setSelectedIndex] = useState<number | null>()
+
+  const { addNewUserTask } = useContext(UserTaskContext)
+  const { openTaskAddedModal } = useContext(TaskAddedModalContext)
+
+  useFocusEffect(
+    useCallback(() => {
+      setSelectedIndex(null)
+    }, [])
+  )
+
+  // @ts-ignore
+  const category = route.params.category as Category
```

Comment:
are you passing objects in the route params to avoid having to fin() by id ?
if yes store the categories in a Map<categoryId, category> and pass only categoryId for example

Reply:
Before it was like you said, but it's more simple to pass the category to the props to avoid a call to the context :)

Reply:
I updated the Typing :)